mpmp/utilities/analysis_utilities.py

```diff
@@ -71,6 +72,112 @@ def load_compressed_prediction_results(results_dir, experiment_descriptor):
             results_df = pd.concat((results_df, id_results_df))
     return results_df
 
+
+def load_purity_binarized_results(results_dir):
+    """Load results of tumor purity prediction experiments.
+
+    Labels are binarized into above/below median.
+
+    Arguments
+    ---------
+    results_dir (str): directory containing results files
+
+    Returns
+    -------
+    results_df (pd.DataFrame): results of classification experiments
+    """
+    results_df = pd.DataFrame()
+    results_dir = Path(results_dir)
+    for results_file in results_dir.iterdir():
+        if not results_file.is_file(): continue
+        results_filename = str(results_file.stem)
+        if 'classify' not in results_filename: continue
+        if results_filename[0] == '.': continue
+        id_results_df = pd.read_csv(results_file, sep='\t')
+        if check_compressed_file(results_filename):
+            id_results_df.training_data += '_compressed'
+        results_df = pd.concat((results_df, id_results_df))
```

Comment:
In case you have performance issues (not sure how big is the data), I think it's more efficient to add all dataframes to a list, and then perform the `pd.concat` at the end.

You don't need to change this. It's just a comment in case you run into performance issues.

Reply:
Good to keep in mind! This doesn't seem to be too slow, but for future code I'll definitely do what you're suggesting.